tests/aead_tests.rs

```diff
@@ -80,6 +82,89 @@ fn aead_chacha20_poly1305() {
         open_with_less_safe_key,
         test_file!("aead_chacha20_poly1305_tests.txt"),
     );
+    test_aead_separate_tag(
+        &aead::CHACHA20_POLY1305,
+        test_file!("aead_chacha20_poly1305_tests.txt"),
+    );
+}
+
+fn test_aead_separate_tag(aead_alg: &'static aead::Algorithm, test_file: test::File) {
```

Comment:
This function can be more similar to `test_aead()`. There's some code duplication from testing `LessSafeKey` as well as `SealingKey`/`OpeningKey`.

Might even make sense to merge this into `test_aead()` by adding `seal_separate_tag` and `open_separate_tag` as parameters.

Reply:
I have changed it to take functions like `test_aead`.